File: services/business-services/menu-v3/application/service/menu_option_service.go
package service

import (
	"menu-v3/domain/model/request"
	"menu-v3/domain/model/response"
)

type MenuOptionService interface {
	FindAll() ([]response.MenuOptionResponse, error)
	FindById(id uint) (*response.MenuOptionResponse, error)
	Save(request request.MenuOptionRequest) (*response.MenuOptionResponse, error)
	Update(request request.MenuOptionRequest, id uint) (*response.MenuOptionResponse, error)
	Delete(id uint) error
}

File: services/business-services/menu-v1/src/main/java/com/demo/bbq/business/menu/application/service/MenuOptionService.java
package com.demo.bbq.business.menu.application.service;

import com.demo.bbq.business.menu.domain.model.request.MenuOptionSaveRequest;
import com.demo.bbq.business.menu.domain.model.request.MenuOptionUpdateRequest;
import com.demo.bbq.business.menu.domain.model.response.MenuOption;
import java.util.List;

public interface MenuOptionService {

  List<MenuOption> findByCategory(String categoryCode);

  MenuOption findByProductCode(String productCode);

  void save (MenuOptionSaveRequest menuOption);

  void update(String productCode, MenuOptionUpdateRequest menuOption);

  void deleteByProductCode(String productCode);
}
